lib/hanami/events/subscriber.rb

```diff
@@ -1,20 +1,14 @@
-require 'hanami/events/handler_runner'
-
 module Hanami
   module Events
-    class Subscriber
-      def initialize(event_name, event_handler, logger = nil)
-        @event_name = event_name
-        @pattern_matcher = Matcher.new(event_name)
-        @handler_runner = HandlerRunner.new(event_handler, logger)
-      end
-
-      def call(event_name, payload)
-        @handler_runner.(payload) if @pattern_matcher.match?(event_name)
+    module Subscriber
```

Comment:
`Subscriber` looks good to me because it's more general, and usage in _class_ looks good:

```ruby
class UseCase
  include Hanami::Events::Subscriber

  subscribe_to event_bus, 'user.created'

  def call(payload)
    # implementation
  end
end
```

`ObjectSubscriber` would be move verbose.